### PR TITLE
Fix memory leak in AppDomainWrapper

### PR DESCRIPTION
**Pull request by snakefoot**
Unsubscribe from AppDomain when closing down

Possible solution for fixing  #1689

---
This change is [Reviewable](https://reviewable.io/reviews/nlog/nlog/1690)

### CODE REVIEW

File: src/NLog/Internal/Fakeables/AppDomainWrapper.cs [hunk 1]
```diff
@@ -102,23 +100,19 @@ public AppDomainWrapper(AppDomain appDomain)
         /// <summary>
         /// Process exit event.
         /// </summary>
-        public event EventHandler<EventArgs> ProcessExit;
+        public event EventHandler ProcessExit
```
Comment (304NotModified) [inline src/NLog/Internal/Fakeables/AppDomainWrapper.cs]: Do you know of this is a breaking change? I think it is (binary breaking change)

**Comment by 304NotModified** [comment]
thanks!

File: src/NLog/Internal/Fakeables/AppDomainWrapper.cs (hunk 1 above)
Comment (snakefoot) [inline src/NLog/Internal/Fakeables/AppDomainWrapper.cs]: Don't know if it is breaking change, but AppDomainWrapper resides within NLog/Internals, so would be surprised if anyone used it.

Alternative solution would be to make AppDomainWrapper IDisposable and then ensure to Dispose()  everywhere.

Comment (304NotModified) [inline src/NLog/Internal/Fakeables/AppDomainWrapper.cs]: Its public but that's mistake IMO. Maybe we should fix that also.

Comment (304NotModified) [inline src/NLog/Internal/Fakeables/AppDomainWrapper.cs]: https://github.com/NLog/NLog/pull/1691

not sure what do to with this

**Comment by 304NotModified** [comment]
close/reopen to trigger AppVeyor

**Pull request merged**